contrib/platform/src/com/sun/jna/platform/win32/Psapi.java

```diff
@@ -320,4 +334,97 @@ class PERFORMANCE_INFORMATION extends Structure {
         public DWORD ThreadCount;
     }
+
+    @FieldOrder({"Flags", "Data"})
+    class PSAPI_WORKING_SET_EX_BLOCK extends Structure {
+        public ULONG_PTR Flags;
+        public ULONG[] Data = new ULONG[Native.POINTER_SIZE == 8 ? 1 : 2];
+        private long innerValue;
+
+        @Override
+        public void read() {
+            super.read();
+            innerValue = this.Data[0].longValue();
+        }
+
+        /**
+         * If this bit is 1, the subsequent members are valid; otherwise they should be ignored.
+         */
+        public boolean Valid() {
+            return getBitFieldValue(1, 0) == 1;
+        }
+
+        /**
+         * The number of processes that share this page. The maximum value of this member is 7.
+         */
+        public int ShareCount() {
+            return getBitFieldValue(3, 1);
+        }
+
+        /**
+         * The memory protection attributes of the page. For a list of values see below.
+         * @see <a href="https://docs.microsoft.com/en-us/windows/desktop/Memory/memory-protection-constants">Memory Protection Constants</a>.
+         */
+        public int Win32Protection() {
+            return getBitFieldValue(11, 3 + 1);
+        }
+
+        /**
+         * If this bit is 1, the page can be shared.
+         */
+        public boolean Shared() {
+            return getBitFieldValue(1, 11 + 3 + 1) == 1;
+        }
+
+        /**
+         * The NUMA node. The maximum value of this member is 63.
+         */
+        public int Node() {
+            return getBitFieldValue(6, 1 + 11 + 3 + 1);
+        }
+
+        /**
+         * If this bit is 1, the virtual page is locked in physical memory.
+         */
+        public boolean Locked() {
+            return getBitFieldValue(1, 6 + 1 + 11 + 3 + 1) == 1;
+        }
+
+        /**
+         * If this bit is 1, the page is a large page.
+         */
+        public boolean LargePage() {
+            return getBitFieldValue(1, 1 + 6 + 1 + 11 + 3 + 1) == 1;
+        }
+
+        /**
+         * If this bit is 1, the page is has been reported as bad.
+         */
+        public boolean Bad() {
+            return getBitFieldValue(1,1 + 1 + 1 + 6 + 1 + 11 + 3 + 1) == 1;
+        }
+
+        /**
+         * Returns innerValue after shifting the value rightShiftAmount, and applying a Bit Mask of size maskLength. Example, <br/>
+         * innerValue = 0011<br/> getBitFieldValue(2, 1) = 0011 >> 1 & 11 = 01
+         * @param maskLength Size of the Bit Mask
+         * @param rightShiftAmount Amount to Shift innerValue to the right by
+         * @return innerValue with the mask and shift applied.
+         */
+        private int getBitFieldValue(final int maskLength, final int rightShiftAmount) {
+            long bitMask = 0;
+
+            for (int l = 0; l <= maskLength - rightShiftAmount; l++) {
+                bitMask |= 1 << l;
+            }
+            return (int) ((innerValue >> rightShiftAmount) & bitMask);
```

Comment:
This isn't right, the bitmask is on the higher bits per lines 417-419, but you are shifting to the lowest bits before masking.  You need to either:
 - shift first and then mask the lowest "length" bits, or
 - mask the specific bits and then shift.  Using the `>>>` operator will save you the need to do special treatment

Reply:
Okay, I think this next iteration will be correct.
```java
private int getBitFieldValue(final int maskLength, final int rightShiftAmount) {
    long bitMask = 0;
                       // Having this Conditional be <= was going give the mask an extra bit.
    for (int l = 0; l < maskLength; l++) {
        bitMask |= 1 << l;
    }
    return (int) ((innerValue >>> rightShiftAmount) & bitMask);
}
```
Now if `maskLength = 4` and `rightShiftAmount = 3`
```
innerValue = 101100111000 // Assume Leading Ones so the Unsigned Shift Prevents Sign Flipping
// Construct Mask
1111
// Shift
101100111000 >> 3 = 101100111
//Apply Mask
101100111 &
000001111 
------------
000000111
```

Reply:
Yes, I agree if you are shifting first before applying the bitmask, then you are using a 0-indexed array and `<` is appropriate.  Ultimately you probably should have tested this (and I will when you commit) with some debug output to convince yourself it works.